Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyManagerComponent.cpp

```diff
@@ -263,19 +263,19 @@ namespace AzToolsFramework
                         return true;
                     },
                     handlerType);
-                if (classes.empty())
+                for (auto cls = classes.begin(); cls != classes.end(); ++cls)
                 {
-                    return pHandlerFound;
+                    pHandlerFound = ResolvePropertyHandler(handlerName, (*cls)->m_typeId);
+                    if (pHandlerFound)
+                    {
+                        return pHandlerFound;
+                    }
                 }
-                else
+                if (const auto* genericInfo = sc->FindGenericClassInfo(handlerType))
```

Comment:
This needs to be specialized for `GetAssetClassId`

Reply:
That would only happen if someone registered a handler with the id equal to whatever `GetGenericClassInfoVectorTypeId()` returns. That actually seems like it could be useful in other contexts. Or should we just keep on special casing `GetAssetClassId()`?

Reply:
`FindGenericClassInfo` accepts the template specialization ID, not the generic ID. Passing in the GenericClassInfo TypeId wouldn't ever locate the genericInfo.

The if block currently only works for the `GetAssetClassId()` scenario, because of the special case code in AssetSerializer.h to register [GenericClassInfo](https://github.com/o3de/o3de/pull/9279/files#diff-f04464db71c462bd0ce78e7af436e1c08c7170e88d749e736eee4e038f7bb978R177) using the `GetAssetClassId()`.

This code should be fine.